File: JavaScript/bank_account.js
class BankAccount {
    constructor() {
      this.amount = 0;
      this.isClosed = true;
    }
  
    open() {   
      if(this.isClosed == false) {
        throw new ValueError();
      }
      
      this.amount = 0;
      this.isClosed = false;
    }
  
    close() {
      if(this.isClosed) {
        throw new ValueError();
      }
  
      this.isClosed = true;
    }
  
    deposit(amount) {
      if(this.isClosed) {
        throw new ValueError();
      }
      else if(amount < 0) {
        throw new ValueError();
      }    

      this.amount += amount;
    }
  
    withdraw(amount) {
      if(this.isClosed) {
        throw new ValueError();
      }
      else if(amount <= 0) {
        throw new ValueError();
      }   
      else if((this.amount - amount) < 0) {
        throw new ValueError();
      }   

      this.amount -= amount;
    }
  
    get balance() {
      if(this.isClosed) {
        throw new ValueError();
      }
      
      return this.amount;
    }
  }
  
class ValueError extends Error {
    constructor() {
      super('Bank account error');
    }
}

const account = new BankAccount();
account.open();
account.deposit(100);
account.withdraw(80);

console.log(account);

File: C#/Bank Account/BankAccount.cs
using System;

public class BankAccount
{
    private decimal _balance;
    private bool _isOpen;

    public BankAccount()
    {
        this._balance = 0;
        this._isOpen = false;
    }

    public void Open()
    {
        this._balance = 0;
        this._isOpen = true;
    }

    public void Close()
    {
        this._balance = 0;
        this._isOpen = false;
    }

    public decimal Balance
    {
        get
        {
            if(this._isOpen == false)
            {
                throw new System.InvalidOperationException();
            }

            return this._balance;
        }
    }

    public void UpdateBalance(decimal change)
    {
        if (this._isOpen == false)
        {
            return;
        }

        this._balance += change;
    }
}